src/ts/sources.ts

```diff
@@ -0,0 +1,27 @@
+import { writable, Writable } from 'svelte/store';
+import { paramsTabId, paramsFrameId } from '../js/constants';
+
+export function twitchSource(): Writable<Ltl.Message | null> {
+  if (paramsTabId == null || paramsFrameId == null) return writable(null);
+
+  const portName = `${Date.now()}${Math.random()}`;
+
+  // MV3 don't support background scripts
+  const port = chrome.tabs.connect(
```

Comment:
`chrome.tabs` doesn't seem to be available when launching embed mode which makes embed mode not work on firefox
![undefined chrome.tabs](https://user-images.githubusercontent.com/50760816/150433026-35ec1ce1-1d84-4a1d-98fc-e0c90e59a198.png)